### ISSUE TITLE
Timeout for cross_validation

### ISSUE THREAD

**Issue by MPiunti** [facebook/prophet#2348]
Would it be feasible a timeout management inside cross validation?

Running prophet in spark for thousands of parallel group of timeseries, i see that for certain groups cross validation is quite expensive
So the idea to set a timeout for each forecast, to be handled  cross validation i.e.  

```
try:
   ....
   df_cv = cross_validation(model, cutoffs=dt_ct,  horizon='14 days' ,   parallel=None,
                                 timeout= '60 sec')
except Exception as e:
  use default params
  
```
is there any trick to  mimic this ?

thanks

**Comment by tcuongd** [comment]
I don't think this time management would live in Prophet, we just provide the functionality rather than handle specific requirements. I'd suggest minimising the number of CV folds by limiting the training period or the window size.
